pkg/ingester/limiter.go

```diff
@@ -222,22 +218,37 @@ func (l *Limiter) convertGlobalToLocalLimit(userID string, globalLimit int) int
 		return 0
 	}
 
+	// Global limit is equally distributed among all the active zones.
+	// The portion of global limit related to each zone is then equally distributed
+	// among all the ingesters belonging to that zoneInside of each zone,
+	zoneCount := l.getZonesCount()
```

Comment:
To reduce likelihood of race conditions, I would call `getZonesCount()` only once and then use in the rest of the logic.